### PR TITLE
send artifact name & device type in update check requests

### PR DESCRIPTION
**Pull request by bboozzoo**
https://tracker.mender.io/browse/MEN-768

@kacf @pasinskim @GregorioDiStefano

### CODE REVIEW

File: client/client_update.go [hunk 1]
```diff
@@ -160,8 +170,20 @@ func processUpdateResponse(response *http.Response) (interface{}, error) {
...
 
-func makeUpdateCheckRequest(server string) (*http.Request, error) {
-	url := buildApiURL(server, "/deployments/device/deployments/next")
+func makeUpdateCheckRequest(server string, current CurrentUpdate) (*http.Request, error) {
+	vals := url.Values{}
+	if current.DeviceType != "" {
+		vals.Add("device_type", current.DeviceType)
+	}
+	if current.Artifact != "" {
+		vals.Add("artifact", current.Artifact)
```
Comment (kacf) [inline client/client_update.go]: Should be `artifact_name`.

File: client/test/server.go [hunk 2]
```diff
@@ -302,6 +304,14 @@ func (cts *ClientTestServer) statusReq(w http.ResponseWriter, r *http.Request, i
 	w.WriteHeader(http.StatusNoContent)
 }
 
+func urlQueryToCurrentUpdate(vals url.Values) client.CurrentUpdate {
+	cur := client.CurrentUpdate{
+		Artifact:   vals.Get("artifact"),
```
Comment (kacf) [inline client/test/server.go]: `artifact_name`

File: client/client_update.go (hunk 1 above)
Comment (bboozzoo) [inline client/client_update.go]: No, it's `artifact` according to the spec https://tracker.mender.io/browse/MEN-744

Comment (kacf) [inline client/client_update.go]: That's wrong, https://tracker.mender.io/browse/MEN-744 updated.

Comment (bboozzoo) [inline client/client_update.go]: done

File: client/test/server.go (hunk 2 above)
Comment (bboozzoo) [inline client/test/server.go]: done

**Comment by kacf** [comment]
Looks good. I doubt this affects those tests, but running it once through Jenkins anyway.

**Pull request merged**